src/System.Text.Json/src/System/Text/Json/JsonElement.cs

```diff
@@ -0,0 +1,372 @@
+                {
+                    throw new KeyNotFoundException();
+                }
+
+                return value;
+            }
+        }
+
+        public JsonElement this[ReadOnlySpan<byte> utf8PropertyName]
+        {
+            get
+            {
+                if (_parent == null)
+                {
+                    throw new InvalidOperationException();
+                }
+
+                if (!_parent.TryGetNamedPropertyValue(_idx, utf8PropertyName, out JsonElement value))
+                {
+                    throw new KeyNotFoundException();
+                }
+
+                return value;
+            }
+        }
+
+        public int GetArrayLength()
+        {
+            if (_parent == null)
+            {
+                throw new InvalidOperationException();
+            }
+
+            return _parent.GetArrayLength(_idx);
+        }
+
+        public bool TryGetProperty(string propertyName, out JsonElement value)
+        {
+            if (propertyName == null)
+                throw new ArgumentNullException(nameof(propertyName));
+
+            return TryGetProperty(propertyName.AsSpan(), out value);
+        }
+
+        public bool TryGetProperty(ReadOnlySpan<char> propertyName, out JsonElement value)
+        {
+            if (_parent == null)
+            {
+                throw new InvalidOperationException();
+            }
+
+            return _parent.TryGetNamedPropertyValue(_idx, propertyName, out value);
+        }
+
+        public bool TryGetProperty(ReadOnlySpan<byte> utf8PropertyName, out JsonElement value)
+        {
+            if (_parent == null)
+            {
+                throw new InvalidOperationException();
+            }
+
+            return _parent.TryGetNamedPropertyValue(_idx, utf8PropertyName, out value);
+        }
+
+        /// <summary>
+        ///   Attempts to get the unprocessed memory contributing to the value of this element.
+        /// </summary>
+        /// <param name="rawValue">
+        ///   Receives the unprocessed memory contributing to the value of this element.
+        /// </param>
+        /// <returns>
+        ///   <c>true</c> if the parent <see cref="JsonDocument"/> was built from UTF-8 data,
+        ///   and the entirety of the value for this element is in contiguous memory.
+        /// </returns>
+        /// <seealso cref="Utf8JsonReader.ValueSpan"/>
+        public bool TryGetRawValue(out ReadOnlyMemory<byte> rawValue)
+        {
+            if (_parent == null)
+            {
+                throw new InvalidOperationException();
+            }
+
+            // This method is try, but calls a non-try, because in the future
+            // JsonDocument and JsonElement may support non-contiguous and/or
+            // differently-encoded data.  JsonDocument's version of this method
+            // would likewise become Try when that happens.
```

Comment:
Meaning a Try variant would be added, and the non-Try would throw in cases where Try would return false?

Reply:
Meaning JsonDocument's GetRawValue would become TryGetRawValue.  (There'd be no callers of the non-Try version)